### PR TITLE
Add support of Bearer Basic Authorization to Ruby client

### PR DESCRIPTION
**Pull request by bopm**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Add support of Bearer Basic Authorization to Ruby client.

### CODE REVIEW

**Comment by bopm** [comment]
cc @cliffano @zlx @autopp

File: modules/openapi-generator/src/main/resources/ruby-client/README.mustache [hunk 1]
```diff
@@ -66,9 +66,15 @@ require '{{{gemName}}}'
...
-  config.password = 'YOUR PASSWORD'{{/isBasic}}{{#isApiKey}}
+{{^isBasicBearer}}
+# Configure HTTP basic authorization: {{{name}}}
+config.username = 'YOUR_USERNAME'
+  config.password = 'YOUR_PASSWORD'
+{{/isBasicBearer}}
+{{#isBasicBearer}}
+# Configure Bearer authorization{{#bearerFormat}} ({{{.}}}){{/bearerFormat}}: {{{name}}}
+config.access_token = 'YOUR_BEARER_TOKEN'
+{{/isBasicBearer}}{{/isBasic}}{{#isApiKey}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/ruby-client/README.mustache]: @bopm thanks for the enhancements. I wonder if you can indent the code above with additional 2-spaces (except line 72)

File: modules/openapi-generator/src/main/resources/ruby-client/configuration.mustache [hunk 2]
```diff
@@ -199,13 +199,27 @@ module {{moduleName}}
...
 {{#isBasic}}
-        '{{name}}' =>
-          {
-            type: 'basic',
-            in: 'header',
-            key: 'Authorization',
-            value: basic_auth_token
-          },
+{{^isBasicBearer}}
+            '{{name}}' =>
```
Comment (autopp) [inline modules/openapi-generator/src/main/resources/ruby-client/configuration.mustache]: This indent seems to be wrong. Can you fix it?
https://github.com/OpenAPITools/openapi-generator/pull/2856/files#diff-c6b80b9611c8982577f1c2ceaeacd8e5R213

File: modules/openapi-generator/src/main/resources/ruby-client/README.mustache (hunk 1 above)
Comment (bopm) [reply modules/openapi-generator/src/main/resources/ruby-client/README.mustache]: Done.

**Comment by bopm** [comment]
All formatting issues are covered, but I am struggle to find a way to remove empty lines in generated files.

File: modules/openapi-generator/src/main/resources/ruby-client/README.mustache [hunk 3]
```diff
@@ -66,9 +66,12 @@ require '{{{gemName}}}'
 {{#apiInfo}}{{#apis}}{{#-first}}{{#operations}}{{#operation}}{{#-first}}{{#hasAuthMethods}}
 # Setup authorization
 {{{moduleName}}}.configure do |config|{{#authMethods}}{{#isBasic}}
+{{^isBasicBearer}}
```
Comment (autopp) [inline modules/openapi-generator/src/main/resources/ruby-client/README.mustache]: @bopm 
> I am struggle to find a way to remove empty lines in generated files.

How about changing the line break here?
```diff
$ git diff
diff --git a/modules/openapi-generator/src/main/resources/ruby-client/api_doc.mustache b/modules/openapi-generator/src/main/resources/ruby-client/api_doc.mustache
index ac7aff1d4f..dc163725e1 100644
--- a/modules/openapi-generator/src/main/resources/ruby-client/api_doc.mustache
+++ b/modules/openapi-generator/src/main/resources/ruby-client/api_doc.mustache
@@ -27,8 +27,7 @@ Method | HTTP request | Description
 require '{{{gemName}}}'
 {{#hasAuthMethods}}
 # setup authorization
-{{{moduleName}}}.configure do |config|{{#authMethods}}{{#isBasic}}
-{{^isBasicBearer}}
+{{{moduleName}}}.configure do |config|{{#authMethods}}{{#isBasic}}{{^isBasicBearer}}
   # Configure HTTP basic authorization: {{{name}}}
   config.username = 'YOUR USERNAME'
   config.password = 'YOUR PASSWORD'{{/isBasicBearer}}{{#isBasicBearer}}
diff --git a/samples/openapi3/client/petstore/ruby/docs/FakeApi.md b/samples/openapi3/client/petstore/ruby/docs/FakeApi.md
index 8c0dbc6989..ab79170143 100644
--- a/samples/openapi3/client/petstore/ruby/docs/FakeApi.md
+++ b/samples/openapi3/client/petstore/ruby/docs/FakeApi.md
@@ -547,7 +547,6 @@ Fake endpoint to test group parameters (optional)
 require 'petstore'
 # setup authorization
 Petstore.configure do |config|
-
   # Configure Bearer authorization (JWT): bearer_test
   config.access_token = 'YOUR_BEARER_TOKEN'
 end
```

Comment (bopm) [reply modules/openapi-generator/src/main/resources/ruby-client/README.mustache]: Seems like it did it. I guess I was looking on subset of changes last time I tried that yesterday. Anyway, it's fine now, thanks.

**Comment by bopm** [comment]
TravisCI is done, just not updating success here for some reason.

**Comment by wing328** [comment]
Right. The job is green: https://travis-ci.org/OpenAPITools/openapi-generator/builds/532295927?utm_source=github_status&utm_medium=notification

**Comment by wing328** [comment]
@bopm thanks for the PR 👍

**Comment by bopm** [comment]
@wing328 looking forward to bring more if ever face same kind of problems. Thanks for all your work on this project!

**Comment by wing328** [comment]
@bopm thanks again for your contribution, which has been included in the v4.0.1 release (https://twitter.com/oas_generator/status/1135534738658062336)

**Comment by tonywok** [comment]
@bopm Hey there, new to open api.

I'd like to take advantage of this feature, but I can't seem to get the generator to produce a client that has the desired auth settings (bearer token).

I'm using a version > v4.0.1. My open api spec is open api version 2.0 -- is this only available in 3.0?

**Comment by wing328** [comment]
JWT Bearer is only available in OAS v3.0 but not v2.0.

Ref: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#jwt-bearer-sample

**Pull request merged**